### PR TITLE
Add LWO3 loader and example

### PR DESCRIPTION
**Pull request by looeee**
### Loader for Lightwave LWO V3 files.

This loader should be largely backwards compatible with LWO2 files, but I haven't tested this.

Here's the [LWO3 spec](http://static.lightwave3d.com/sdk/2018/html/filefmts/lwo3.html). It's _somewhat_ accurate. Nearly everything that's possible to support is supported, with a couple of notable exceptions:

### Issues and future work

* quads are converted to triangles, but higher order polys will log an error if encountered
* conversion from left handed LWO coords to right handed three.js coords was accomplished by flipping `x` -> `-x`, but as a result material `FrontSide` and `BackSide` have been switched. More testing is needed to ensure that this switch is correct in all cases, especially with regards to pivot points
* Standard UV maps (VMAPs) are supported, but discontinuous vertex maps (VMADs) are not. 

The last issue is the biggest remaining problem. LWO specifies UVs in two ways:

1. VMAPs, which correspond with with standard three.js UVs. These are fully supported
2. VMADs, which were added to Lightwave a few releases ago to allow for discontinuities in the mesh (AKA seams). 

VMADs are essentially multiple UV maps per mesh - sometimes lots of them, in the tens or hundreds. The three.js equivalent is duplicated vertices. It will take some work and refactoring of the loader to make these work. 

### Example

I've also added a simple example (see below) with test materials. Justin (@jds580s) from onthez.com sponsored development of this loader, and he's planning to contribute a better model sometime soon. We'll improve the example then.

### CODE REVIEW

File: examples/js/loaders/LWOLoader.js [hunk 1]
```diff
@@ -0,0 +1,2304 @@
...
+
+			var self = this;
+
+			var path = ( self.path === undefined ) ? THREE.LoaderUtils.extractUrlBase( url ) : self.path;
+
+			// give the mesh a default name based on the filename
+			var modelName = url.split( path ).pop().split( '.' )[ 0 ];
+
+			var loader = new THREE.FileLoader( this.manager );
+			loader.setResponseType( 'arraybuffer' );
```
Comment (Mugen87) [inline examples/js/loaders/LWOLoader.js]: It seems you are not using `loader.setPath()` here. This should be done here so the loader is more consistent to the others.

Comment (looeee) [reply examples/js/loaders/LWOLoader.js]: Good catch!

**Comment by looeee** [comment]
A couple of points regarding the test materials from the example:

1. materials in the specular row all look identical, since Lightwave has setttings for specular in their BSFD material, but there's no similar settings in the `MeshStandardMaterial`
2. Materials on the left of the refraction row were incorrectly set in LW to have negative IOR so they look a bit weird. But I guess that means that three.js is ready to render these crazy new [negative IOR metamaterials](https://en.wikipedia.org/wiki/Negative-index_metamaterial)! 😁

**Comment by gkjohnson** [comment]
@looeee Awesome!

I'm not seeing an normal maps or environment maps in the example and in the console it looks like it can't load a few textures. Did you forget to commit `normal_2.jpg`, `BumpMap_Rock.jpg`, and `Env_map_sphere_1.jpg`?

**Comment by looeee** [comment]
Oh, good catch. I've committed the textures. 

Hmm, this reminded me of one other issue with the loader (it's six months since I wrote it so my memory is a little rusty!)

Lightwave expects textures to be in an `Images/` folder relative to the "Content Directory" which you specify in settings. If the images are found there, it stores a relative path. If you use textures from any other folder, it stores an absolute path. 

I haven't tested how this works with `setResourcePath` in the loader, and I won't have time to test that this week. If we merge this PR as is I'll schedule testing this for next week sometime, and we can remove these textures that I've just committed from the repo and use some from the `/examples/textures` folder instead. 

Side note: it's not possible to embed textures in LWO files, they always need to be included separately.

**Comment by Mugen87** [comment]
https://raw.githack.com/looeee/three.js/lwo_loader/examples/webgl_loader_lwo.html

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**